test/reduce_test.cpp

```diff
@@ -747,32 +747,18 @@ struct reduce_driver : test_driver
     std::vector<std::vector<int>> get_toreduce_dims()
     {
         std::vector<std::vector<int>> tensor_dims = {
-            {0}, {1}, {2}, {3}, {0, 1}, {0, 3}, {0, 2}, {2, 3}, {0, 1, 3}, {1, 2, 3}};
+            {0}, {1}, {2}, {3}, {0, 1}, {0, 3}, {0, 2}, {2, 3}, {0, 1, 3}, {1, 2, 3}, {0, 1, 2, 3}};
 
-        if(std::is_same<T, half_float::half>::value)
-        {
-            tensor_dims.push_back({0, 1, 2, 3});
-
-            return tensor_dims;
-        }
-        else
-            return tensor_dims;
+        return tensor_dims;
     }
 
     reduce_driver()
     {
         add(inLengths, "D", generate_data(get_tensor_lengths()));
         add(toReduceDims, "R", generate_data(get_toreduce_dims()));
-        add(reduceOp, "ReduceOp", generate_data({
-                MIOPEN_REDUCE_TENSOR_ADD, MIOPEN_REDUCE_TENSOR_MIN, MIOPEN_REDUCE_TENSOR_AMAX,
-                    MIOPEN_REDUCE_TENSOR_AVG
-#if MIOPEN_API_VERSION_REDUCE_TENSOR > 1
-                    ,
-                    MIOPEN_REDUCE_TENSOR_NORM1, MIOPEN_REDUCE_TENSOR_NORM2
-#endif
-            }));
+        add(reduceOp, "ReduceOp", generate_data({0, 4, 5, 6, 7}));
```

Comment:
[Q] Perhaps dumb question, but -- why operations 1, 2 and 3 are not tested?

Reply:
- Take too long to run `#> test_reduce_test --all`
- For MUL, the values could easily reduce to "zero"  and MIOpen will report failure when both outputs (host & gpu) are all zeroes. 
- MIN & MAX are actually covered by AMAX

Reply:
Also, in some case, when doing all-reduction of ADD to reduce a big tensor to one value, it is also easy to get a reduced result of one zero value.  This case is avoided by me.  Later  I will try to improve the mean of the initialized data to be a little above 0.0f so that this could not occur

Reply:
> Take too long to run

This is not a problem. Some convolution tests take _hours_. 10-15 minutes is absolutely ok. Now it takes < 5 minutes.

Reply:
If it is possible to test all ops, then this should be done. Of course, this is a bit out of formal scope of this PR (except operation 2 which was tested but now not). What do you think?